java/BUILD

```diff
@@ -123,50 +134,87 @@ DISTRO_VERSIONS = {
         "",
         "_debug",
     ]
+    for user in [
+        "_root",
+        "_nonroot",
+    ]
     for distro_suffix in DISTRO_SUFFIXES
 ]
 
-container_test(
-    name = "java8_debian9_test",
-    configs = ["testdata/java8.yaml"],
-    image = ":java8_debian9",
-    tags = [
-        "amd64",
-        "manual",
-    ],
-)
+[
+    container_test(
+        name = "java8_debian9" + user + "_test",
+        configs = ["testdata/java8.yaml"],
+        image = ":java8" + user + "_debian9",
+        tags = [
+            "amd64",
+            "manual",
+        ],
+    )
+    for user in [
+        "_root",
+        "_nonroot",
+    ]
+]
 
-container_test(
-    name = "java8_debug_debian9_test",
-    configs = ["testdata/java8_debug.yaml"],
-    image = ":java8_debug_debian9",
-    tags = [
-        "amd64",
-        "manual",
-    ],
-)
+[
+    container_test(
+        name = "java8_debug" + user + "_debian9_test",
+        configs = ["testdata/java8_debug.yaml"],
+        image = ":java8_debug" + user + "_debian9",
+        tags = [
+            "amd64",
+            "manual",
+        ],
+    )
+    for user in [
+        "_root",
+        "_nonroot",
+    ]
+]
 
-[container_test(
-    name = "java11" + distro_suffix + "_test",
-    configs = ["testdata/java11" + distro_suffix + ".yaml"],
-    image = ":java11" + distro_suffix,
-    tags = [
-        "amd64",
-        "manual",
-    ],
-) for distro_suffix in DISTRO_SUFFIXES]
+[
+    container_test(
+        name = "java11" + user + distro_suffix + "_test",
+        configs = ["testdata/java11" + distro_suffix + ".yaml"],
+        image = ":java11" + user + distro_suffix,
+        tags = [
+            "amd64",
+            "manual",
+        ],
+    )
+    for user in [
+        "_root",
+        "_nonroot",
+    ]
```

Comment:
Sorry, but we prefer applying a consistent pattern as in `cc/BUILD`:

https://github.com/GoogleContainerTools/distroless/blob/40976b8b65aec783a7571ad20d7b1fa0cd12c69e/cc/BUILD#L24-L27

That is, no underscore for `user`.